src/torchjd/autojac/_utils.py

```diff
@@ -30,3 +32,72 @@ def _check_retain_graph_compatible_with_chunk_size(
             "When using `retain_graph=False`, parameter `parallel_chunk_size` must be `None` or "
             "large enough to compute all gradients in parallel."
         )
+
+
+def _traverse_ag_graph(grad_graph: Node) -> list[Tensor]:
+    """
+    Traverses an autograd graph recursively and extracts all variables (trainable parameters) that lead to the node.
+    @param grad_graph: autograd computation node
+    @return: found params of calculation
+    """
+    child_nodes = grad_graph.next_functions
+    output_params = []
+    for child in child_nodes:
+        if child[0] is None:
+            # constants in calculation case
+            continue
+        if hasattr(child[0], "variable"):
+            output_params.append(child[0].variable)
+        else:
+            output_params += _traverse_ag_graph(child[0])
```

Comment:
Refactored it into a non recursive implementation. Didnt know that about python actually. I agree with you especially if the current trend of deeper networks continues this will become a problem later anyways